src/Apps/IdentityVerification/IdentityVerificationApp/StartIdentityVerification.tsx

```diff
@@ -0,0 +1,168 @@
+import { ErrorModal } from "Components/Modal/ErrorModal"
+import React, { useState } from "react"
+import { RelayProp } from "react-relay"
+import { useTracking } from "react-tracking"
+import { commitMutation, graphql } from "relay-runtime"
+import createLogger from "Utils/logger"
+
+const logger = createLogger("StartIdentityVerification.tsx")
+
+interface Props {
+  relay: RelayProp
+  identityVerification: {
+    internalID: string
+  }
+}
+
+export const StartIdentityVerification: React.FC<Props> = ({
+  identityVerification,
+  relay: { environment },
+}) => {
+  const [requesting, setRequesting] = useState(false)
+  const [showErrorModal, setShowErrorModal] = useState(false)
+  const { trackEvent } = useTracking()
+
+  const trackClickedContinueToVerification = () => {
+    trackEvent({
+      context_page_owner_id: identityVerification.internalID,
+      action_type: Schema.ActionType.ClickedContinueToIdVerification,
+      context_page: Schema.PageName.IdentityVerificationPage,
+    })
+  }
+
+  async function startIdentityVerification() {
+    const mutation = new Promise<string>((resolve, reject) => {
+      commitMutation<StartIdentityVerificationMutation>(environment, {
+        mutation: graphql`
+          mutation StartIdentityVerificationMutation(
+            $input: startIdentityVerificationMutationInput!
+          ) {
+            startIdentityVerification(input: $input) {
+              startIdentityVerificationResponseOrError {
+                ... on StartIdentityVerificationSuccess {
+                  identityVerificationFlowUrl
+                }
+                ... on StartIdentityVerificationFailure {
+                  mutationError {
+                    detail
+                    error
+                    message
+                  }
+                }
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { identityVerificationId: identityVerification.internalID },
+        },
+        onError: reject,
+        onCompleted: (response, errors) => {
+          if (errors && errors.length > 0) {
+            reject(new Error(JSON.stringify(errors)))
+          } else {
+            const {
+              startIdentityVerification: {
+                startIdentityVerificationResponseOrError: {
+                  identityVerificationFlowUrl,
+                  mutationError,
+                },
+              },
+            } = response
+            if (mutationError) {
+              reject(new Error(JSON.stringify(mutationError)))
+            } else {
+              resolve(identityVerificationFlowUrl)
+            }
+          }
+        },
+      })
+    })
+
+    mutation
+      .then(handleMutationSuccess)
+      .catch(handleMutationError)
+      .finally(() => {
+        setRequesting(false)
+      })
+  }
+
+  const handleMutationSuccess = (identityVerificationFlowUrl: string) => {
+    location.assign(identityVerificationFlowUrl)
```

Comment:
We should probably use `props.router` or `props.location` rather than relying on the global one.